pkg/plugins/bootstrap/k8s/plugin.go

```diff
@@ -179,6 +220,18 @@ func makeOldLockAnnotation() string {
 	return string(annotJson)
 }
 
+// startLeaderComponents adds all components that need leader election to the
+// Manager. The Manager should be running already, any components added after
+// that are started immediately.
+func (cm *kubeComponentManager) startLeaderComponents() {
```

Comment:
I think any error in this function should be returned, logged and then `os.Exit(1)`.

Reply:
Good point, maybe we should return an `error` here and then return that error in `Start(done <-chan struct{}) error` further down? That should trigger `kuma-cp` to exit.

Reply:
Yes, if it's possible to get rid of `os.Exit(1)`, that's even better.

Reply:
Done. Used `golang.org/x/sync`. It's also possible with just channels, `errgroup` is nicer to read I think.